spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkV2Filters.java

```diff
@@ -245,6 +230,57 @@ public static Expression convert(Predicate predicate) {
     return null;
   }
 
+  private static <T> UnboundPredicate<T> handleEQPredicate(Predicate predicate) {
```

Comment:
It should be more accurate with `handleEQOrNotEQPredicate` or `handleEqualOrNotEqualPredicate`. However, the name is a little stranger or too long.

Reply:
Sorry I was probably not so clear.

Glad to see the lambda removed but I think its still not clean that there's a repeated if/else in the common code (the predicate EQ or NOT_EQ is checked twice, in case statement and then common method).  So how about just make sure each layer checks once.

```

case EQ:
   PredicateValues predicateValues = predicateValues(predicate);
   handleEqual(predicateValues);
case NOT_EQ:  
   PredicateValues predicateValues = predicateValues(predicate);
   return handleNotEqual(predicateValues);
 
class PredicateValues {
  Object value;
  Column column;
}

// predicateValues returns if one side is literal and other is column, null otherwise
// handle[Not]Equal returns null if predicateValues is null
```